test/rfxcom.spec.js

```diff
@@ -4,164 +4,175 @@ var rfxcom = require('../lib'),
     matchers = require('./matchers'),
     protocols = rfxcom.protocols;
 
-describe("RfxCom", function() {
-    beforeEach(function() {
+describe("RfxCom", function () {
+    beforeEach(function () {
         this.addMatchers({
             toHaveSent: matchers.toHaveSent
         });
     });
 
-    describe("RfxCom class", function() {
-        describe("data event handler", function() {
-            it("should emit a response message when it receives message type 0x02", function(done) {
+    describe("RfxCom class", function () {
+        describe("data event handler", function () {
+            it("should emit a response message when it receives message type 0x02", function (done) {
                 var fakeSerialPort = new FakeSerialPort(),
-                    device = new rfxcom.RfxCom("/dev/ttyUSB0", {
+                        device = new rfxcom.RfxCom("/", {
```

Comment:
I now have connect/disconnect handling working nicely on Mac OS X, Linux & Windows 7, all using events. I have also reverted to using all numeric subtypes as you asked.